src/lib.rs

```diff
@@ -1259,13 +1259,11 @@ impl CertificateParams {
 /// (only relevant for CA certificates)
 #[derive(Debug, PartialEq, Eq, Clone)]
 pub struct NameConstraints {
-	/// If non-empty, a whitelist of subtrees that the
-	/// domain has to match.
+	/// If non-empty, a list of subtrees that the domain has to match.
 	pub permitted_subtrees: Vec<GeneralSubtree>,
-	/// A list of excluded subtrees.
+	/// If non-empty, a list of excluded subtrees that the domain must not match.
```

Comment:
I'd rather remove the "If non-empty," prefix from `permitted_subtrees`? At least I'm not sure what value it adds. If you're going to add "that the domain must not match", maybe remove "excluded"? Otherwise this feels like it is trying to say the same thing twice (I agree that the prose explanation is more clear than merely using "excluded").

Reply:
Changed to remove `If non-empty` prefixes, and the "excluded" from the excluded subtrees comment.

Thanks, agreed it reads better :+1: